services/f-droid/f-droid.tester.js

```diff
@@ -0,0 +1,51 @@
+
+t.create('Package is found')
+  .get('/v/org.pacien.tincapp.json')
+  .intercept(nock =>
+    nock('https://f-droid.org')
+      .get('/en/packages/org.pacien.tincapp/')
+      .reply(200, manyVersions)
+  )
+  .expectJSON({ name: 'F-Droid', value: 'v0.19' })
+
+t.create('Package is not found')
+  .get('/v/org.pacien.tincapp.json')
+  .intercept(nock =>
+    nock('https://f-droid.org')
+      .get('/en/packages/org.pacien.tincapp/')
+      .reply(404, manyVersions)
+  )
+  .expectJSON({ name: 'F-Droid', value: 'app not found' })
+
+t.create('The api changed')
+  .get('/v/org.pacien.tincapp.json')
+  .intercept(nock =>
+    nock('https://f-droid.org')
+      .get('/en/packages/org.pacien.tincapp/')
+      .reply(200, '')
+  )
+  .expectJSON({ name: 'F-Droid', value: 'fix this badge' })
+
+t.create('The real api did not change')
+  .get('/v/org.thosp.yourlocalweather.json')
+  /*  .intercept(nock => // uncomment to test the test
+    nock('https://f-droid.org')
+      .get('/en/packages/org.pacien.tincapp/')
+      .reply(200, '')
+  )*/
+  .expectJSONTypes({ name: Joi.string(), value: Joi.string() })
```

Comment:
Would our test validator `isVPlusDottedVersionAtLeastOne` be of any help here? It's used in _eclipse-marketplace.spec.js_ for instance.

Reply:
There are three cases:
1. version is given
2. app not found
3. app found but API changed

We want it to fail in cases  (2) because we then can not detect an API change and (3) the API has changed.
Thus, this `isVPlusDottedVersionAtLeastOne` would be great!